src/wallet/mod.rs

```diff
@@ -454,6 +464,170 @@ impl Wallet {
+        // Ctip.
+        let ctip_foreign_utxo = match sidechain_ctip {
+            Some(sidechain_ctip) => {
+                let outpoint = bdk::bitcoin::OutPoint {
+                    txid: convert::bitcoin_txid_to_bdk_txid(sidechain_ctip.outpoint.txid),
+                    vout: sidechain_ctip.outpoint.vout,
+                };
+
+                let ctip_transaction = self
+                    .fetch_ctip_transaction(sidechain_ctip.outpoint.txid)
+                    .await?;
+
+                let psbt_input = bdk::bitcoin::psbt::Input {
+                    non_witness_utxo: Some(ctip_transaction),
+                    ..bdk::bitcoin::psbt::Input::default()
+                };
+
+                Some((psbt_input, outpoint))
+            }
+            None => None,
+        };
+
+        let (psbt, _) = {
+            let wallet = self.bitcoin_wallet.lock();
+
+            let mut builder = wallet.build_tx();
+
+            builder
+                // important: the M5 OP_DRIVECHAIN output must come directly before the OP_RETURN sidechain address output.
+                .add_recipient(
+                    op_drivechain_output.script_pubkey,
+                    op_drivechain_output.value,
+                )
+                .add_data(&sidechain_address_data);
+
+            if let Some(fee_sats) = fee_sats {
+                builder.fee_absolute(fee_sats.to_sat());
+            }
+
+            if let Some((ctip_psbt_input, outpoint)) = ctip_foreign_utxo {
+                // This might be wrong. Seems to work!
+                let satisfaction_weight = 0;
+
+                builder
+                    .add_foreign_utxo(outpoint, ctip_psbt_input, satisfaction_weight)
+                    .map_err(WalletError)?;
+            }
+
+            builder.finish().into_diagnostic()?
+        };
+
+        Ok(psbt)
+    }
+
+    /// Creates a deposit transaction, persists it to the database, and returns the TXID.
+    /// This is also known as a M5 message, in BIP300 nomenclature.
+    ///
+    /// https://github.com/bitcoin/bips/blob/master/bip-0300.mediawiki#m5----deposit-btc-from-l1-to-l2
+    pub async fn create_deposit(
+        &self,
+        sidechain_number: SidechainNumber,
+        sidechain_address: String,
+        value_sats: Amount,
+        fee_sats: Option<Amount>,
+    ) -> Result<bdk::bitcoin::Txid> {
+        // If this is None, there's been no deposit to this sidechain yet. We're the first one!
+        let sidechain_ctip = self.validator.get_ctip(sidechain_number)?;
+        let sidechain_ctip = sidechain_ctip.as_ref();
+
+        let sidechain_ctip_amount = sidechain_ctip
+            .map(|ctip| ctip.value)
+            .unwrap_or(Amount::ZERO);
+
+        let op_drivechain_output = Self::create_deposit_op_drivechain_output(
+            sidechain_number,
+            sidechain_ctip_amount,
+            value_sats,
+        );
+
+        tracing::debug!(
+            "Created OP_DRIVECHAIN output: {}, {} ",
```

Comment:
Prefer giving explicit names to logged values, ie.
```rust
tracing::debug!(
    "Created OP_DRIVECHAIN output with value `{}`, spk `{}` ",
    ...
)
```

or
```rust
tracing::debug!(
    value = %op_drivechain_output.value,
    spk = %op_drivechain_output.script_pubkey.to_asm_string(),
    "Created OP_DRIVECHAIN output"
)
```

see
https://docs.rs/tracing/latest/tracing/index.html#recording-fields